pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -598,6 +599,6 @@ func newNSTemplateSet(userAcc *toolchainv1alpha1.UserAccount) *toolchainv1alpha1
 }
 
 // ToIdentityName converts the given `userID` into an identity
-func ToIdentityName(userID string) string {
+func ToIdentityName(userID string, config *configuration.Registry) string {
```

Comment:
here you could introduce a specific interface which has a single `GetIdP()` func, so it's more flexible in term of usage if we want to replace the `config` with a custom/mock object. (That's just a suggestion)

Reply:
I think that's a very good suggestion as it would allow us to reuse that function for any number of applications beyond just configuration but it looked a bit funny when I created a small interface that only returns a single string and the interface is only used in this one place at the moment. Do we foresee it being reused elsewhere? If not, should wait until we need to use it somewhere else or want to expand on it?

Reply:
it's just that having a simple interface here makes it easier to test. You could provide a mock that returns a specific value for your test case, without having to deal with the initialization of a full-size configuration object.

Reply:
Good point, I'll make the change

Reply:
Done in 69cd855580034f247065ba24ca5e4ffc673acfeb

Reply:
just one thing that I noticed - we have this fancy new interface but is not used anywhere, so do we really need it? The tests still rely on and use the config without any problem, so I'm missing the benefit of the interface. 

I'm not saying that you should remove it, I'm just sharing my observation :wink: You can keep it if you want :-)